mysqld_exporter.go

```diff
@@ -172,6 +174,65 @@ func init() {
+		return flag && filters[name]
+	}
+	return flag
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	var filters map[string]bool
+	params := r.URL.Query()["collect[]"]
+	log.Debugln("collect query:", params)
+
+	if len(params) > 0 {
+		filters = make(map[string]bool)
+		for _, param := range params {
+			filters[param] = true
+		}
+	}
+
+	collect := collector.Collect{
+		SlowLogFilter:        *slowLogFilter,
+		Processlist:          filter(filters, "info_schema.processlist", *collectProcesslist),
+		TableSchema:          filter(filters, "info_schema.tables", *collectTableSchema),
+		InnodbTablespaces:    filter(filters, "info_schema.innodb_tablespaces", *collectInnodbTablespaces),
+		InnodbMetrics:        filter(filters, "info_schema.innodb_metrics", *collectInnodbMetrics),
+		GlobalStatus:         filter(filters, "global_status", *collectGlobalStatus),
+		GlobalVariables:      filter(filters, "global_variables", *collectGlobalVariables),
+		SlaveStatus:          filter(filters, "slave_status", *collectSlaveStatus),
+		AutoIncrementColumns: filter(filters, "auto_increment.columns", *collectAutoIncrementColumns),
+		BinlogSize:           filter(filters, "binlog_size", *collectBinlogSize),
+		PerfTableIOWaits:     filter(filters, "perf_schema.tableiowaits", *collectPerfTableIOWaits),
+		PerfIndexIOWaits:     filter(filters, "perf_schema.indexiowaits", *collectPerfIndexIOWaits),
+		PerfTableLockWaits:   filter(filters, "perf_schema.tablelocks", *collectPerfTableLockWaits),
+		PerfEventsStatements: filter(filters, "perf_schema.eventsstatements", *collectPerfEventsStatements),
+		PerfEventsWaits:      filter(filters, "perf_schema.eventswaits", *collectPerfEventsWaits),
+		PerfFileEvents:       filter(filters, "perf_schema.file_events", *collectPerfFileEvents),
+		PerfFileInstances:    filter(filters, "perf_schema.file_instances", *collectPerfFileInstances),
+		UserStat:             filter(filters, "info_schema.userstats", *collectUserStat),
+		ClientStat:           filter(filters, "info_schema.clientstats", *collectClientStat),
+		TableStat:            filter(filters, "info_schema.tablestats", *collectTableStat),
+		QueryResponseTime:    filter(filters, "info_schema.query_response_time", *collectQueryResponseTime),
+		EngineTokudbStatus:   filter(filters, "engine_tokudb_status", *collectEngineTokudbStatus),
+		EngineInnodbStatus:   filter(filters, "engine_innodb_status", *collectEngineInnodbStatus),
+		Heartbeat:            filter(filters, "heartbeat", *collectHeartbeat),
+		HeartbeatDatabase:    *collectHeartbeatDatabase,
+		HeartbeatTable:       *collectHeartbeatTable,
+	}
+
+	registry := prometheus.NewRegistry()
+	registry.MustRegister(prometheus.NewGoCollector())
+	registry.MustRegister(prometheus.NewProcessCollector(os.Getpid(), ""))
+	registry.MustRegister(collector.New(dsn, collect))
+	prometheus.DefaultRegisterer = registry
+
+	// Delegate http serving to Promethues client library, which will call collector.Collect.
```

Comment:
Promethues -> Prometheus

Reply:
I copied this typo from another exporter 😄